Ability to post spoilers & Removed padding from toots

EmiMidnight opened the pull request:
First of all, sorry for the other pull request. Still trying to figure out git and I did some mistakes with the first one. Hope you can forgive me.

These changes add the ability for users to add content warnings using mastodons new spoiler system. It adds a new checkbox plus input box to the toot composer and has the newly added API calls.
The next step would be to actually implement content warnings on the timeline itself.

I also went ahead and fixed the weird padding that occurred after most toots. The code now strips the double new line from each toot that has it.
The timeline now looks a lot cleaner.

Commit:
Added the ability to add content warnings to toots using the new spoiler system.
Fixed excessive padding from toots by stripping double new line at the end

charlag commented:
@DRungelrath  sorry for the long silence. Thanks for your PR.
Don't you worry about pull requests, we are all learning.
It looks good to me but I would like to hear from @alin-rautoiu too.

EmiMidnight commented:
@charlag No problem! 👍 
Yeah, working on this has been a really fun learning experience so far actually!

In the meantime I've fixed a few things:
-Some "default" toots had the empty status_tv displaying, adding even more padding. It's gone now.
-Boosts now show inline images like normal toots
-Timeline refresh now works properly. (before, it was putting all the new, updated toots at the end of the timeline)

The latter has introduced a weird bug regarding the isFavorited status (or rather the fav icon itself) though that I can't seem to fix, so I'm still trying out a few things to get it working perfectly.

alin-rautoiu commented:
It looks good. As Mastodon itself grows we must  keep up with it. Thank you for the help/
